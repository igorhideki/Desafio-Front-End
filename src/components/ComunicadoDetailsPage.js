import React, { Component } from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import api from '../requests/apis';
import ComunicadoItemFile from './ComunicadoItemFile';

class ComunicadoDetailsPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      comunicado: {},
      isLoading: false
    };
  }

  componentDidMount() {
    this.setState(() => ({ isLoading: true }));
    api.loadComunicadoById(this.props.match.params.id)
      .then((res) => {
        this.setState({ comunicado: res.data });
        this.setState(() => ({ isLoading: false }));
        return res.data;
      })
      .then((data) => api.updateComunicado(this.props.match.params.id, { ...data, seen: true }))
    ;
  }

  handleCategoriaItem(type) {
    switch (type) {
      case 0:
        return { color: 'comunicados__item--orange', name: 'Informativos SUN' };
      case 1:
        return { color: 'comunicados__item--blue', name: 'Força de Vendas' };
      case 2:
        return { color: 'comunicados__item--green', name: 'Vivo Ligado' };
      case 3:
        return { color: 'comunicados__item--purple', name: 'Turbine Suas Vendas' };
      default:
        return { color: '', name: '' };
    }
  }

  handleFormatDate(date) {
    const parseDate = moment(date, 'YYYY-MM-DD HH:mm');
    return moment(parseDate).format('DD/MM/YYYY - HH[h]mm');
  }

  render() {
    return (
      <div className="container">
        <div className={`comunicados__item comunicados__item--no-pointer ${this.handleCategoriaItem(this.state.comunicado.type).color}`}>
          <div className="comunicados__row">
            <span className="comunicados__title">{this.state.comunicado.title}</span>
            <span className="comunicados__id">{this.state.comunicado.id}</span>
          </div>
          {this.state.comunicado.files && this.state.comunicado.files.length > 0 && this.state.comunicado.files.map((file) => {
            return <ComunicadoItemFile key={file.name} {...file} />;
          })}
          <div className="comunicados__row">
            <span className="comunicados__type">{this.handleCategoriaItem(this.state.comunicado.type).name}</span>
            <span className="comunicados__date">{this.handleFormatDate(this.state.comunicado.date)}</span>
          </div>
          <div className="mt-m">
            {this.state.comunicado.description !== '' && <p className="comunicados__subtitle">Resumo:</p>}
            <p className="comunicados__description">{this.state.comunicado.description}</p>
            {this.state.comunicado.description !== '' && <p className="comunicados__subtitle">Descrição:</p>}
            <p className="comunicados__description">{this.state.comunicado.fullDescription}</p>
          </div>
        </div>
      </div>
    );
  }
}

ComunicadoDetailsPage.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string,
    }),
  }).isRequired,
};

export default ComunicadoDetailsPage;
