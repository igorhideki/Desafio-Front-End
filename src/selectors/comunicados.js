import moment from 'moment';

const getVisiblesComunicados = (comunicados, { type = -1 }) => {
  const formatDate = 'YYYY-MM-DD HH:mm';
  return comunicados.filter((comunicado) => {
    if (type === -1) {
      return comunicado;
    }
    return comunicado.type === type;
  }).sort((a, b) => {
    return moment(a.date, formatDate).format() < moment(b.date, formatDate).format() ? 1 : -1;
  });
};

export default getVisiblesComunicados;

