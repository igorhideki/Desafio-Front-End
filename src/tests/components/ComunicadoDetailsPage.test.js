import React from 'react';
import { mount } from 'enzyme';
import ComunicadoDetailsPage from '../../components/ComunicadoDetailsPage';
import comunicados from '../fixtures/comunicados';

test('should render ComunicadoDetailsPage correctly', () => {
  const comunicado = comunicados[1];
  const match = { params: { id: comunicado.id.toString() } };
  const wrapper = mount(<ComunicadoDetailsPage match={match} />);
  wrapper.setState(() => ({ isLoading: false, comunicado }));
  expect(wrapper).toMatchSnapshot();
});
