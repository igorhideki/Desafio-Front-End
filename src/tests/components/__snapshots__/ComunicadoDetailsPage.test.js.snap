// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ComunicadoDetailsPage correctly 1`] = `
<ComunicadoDetailsPage
  match={
    Object {
      "params": Object {
        "id": "2",
      },
    }
  }
>
  <div
    className="container"
  >
    <ComunicadoDetailsItem
      date="2018-05-02 10:30"
      description="Planos controle 5GB de internet 4G."
      fullDescription="Ligações ilimitadas locais para todas as operadoras. Ligações ilimitadas pra fixos nacional e qualquer Vivo do Brasil com o código 15. SMS ilimitado para qualquer operadora do Brasil."
      id={2}
      seen={false}
      title="Bônus Fevereiro"
      type={1}
    >
      <div
        className="comunicados__item comunicados__item--no-pointer comunicados__item--blue"
      >
        <div
          className="comunicados__row"
        >
          <span
            className="comunicados__title"
          >
            Bônus Fevereiro
          </span>
          <span
            className="comunicados__id"
          >
            2
          </span>
        </div>
        <div
          className="comunicados__row"
        >
          <span
            className="comunicados__type"
          >
            Força de Vendas
          </span>
          <span
            className="comunicados__date"
          >
            02/05/2018 - 10h30
          </span>
        </div>
        <div
          className="mt-m"
        >
          <p
            className="comunicados__subtitle"
          >
            Resumo:
          </p>
          <p
            className="comunicados__description"
          >
            Planos controle 5GB de internet 4G.
          </p>
          <p
            className="comunicados__subtitle"
          >
            Descrição:
          </p>
          <p
            className="comunicados__description"
          >
            Ligações ilimitadas locais para todas as operadoras. Ligações ilimitadas pra fixos nacional e qualquer Vivo do Brasil com o código 15. SMS ilimitado para qualquer operadora do Brasil.
          </p>
        </div>
      </div>
    </ComunicadoDetailsItem>
  </div>
</ComunicadoDetailsPage>
`;
